### ISSUE TITLE
MACE vs ECE

### ISSUE THREAD

**Issue by tfjgeorge** [uncertainty-toolbox/uncertainty-toolbox#46]
Hello, thanks for sharing your code!

I noticed that you implement the metric "MACE", while I was more familiar with the term "ECE". Is there a difference between the two?

**Comment by YoungseogChung** [comment]
Hi, thanks for your interest in the toolbox and for your question.

TL;DR: We consider the two quantities identical. 

As far as we are aware, ECE (expected calibration error) is formally defined in literature only in the *classification* setting ([Guo et al. 2017](https://arxiv.org/pdf/1706.04599.pdf)), whereas the toolbox currently provides utilities for the *regression* setting.
However, there are still some similar concepts in both settings: we can visualize a calibration plot (a.k.a. reliability diagram) and produce a summary statistic that averages the absolute deviation from the (ideal) diagonal line. 
This is what ECE measures in the classification setting (where the axes of the plot are x: confidence, and y: accuracy). 
This is also the quantity that MACE (and [Miscalibration Area](https://github.com/uncertainty-toolbox/uncertainty-toolbox/blob/b2823a1e57af7fe8266a207950deb19126f55a31/uncertainty_toolbox/metrics_calibration.py#L142)) measures (whereas now, the axes are x: expected proportions under quantile and y: observed proportions under quantile).
Therefore, we consider the two quantities (ECE and MACE) identical. 

Let us know if this makes sense to you, or if you have other questions, thanks!

**Comment by tfjgeorge** [comment]
Thanks this makes total sense!
